File: js/programmers/level.1/prgm_176963_1.js
function solution(name, yearning, photo) {
  const obj = {};
  const result = [];

  for (let i = 0; i < name.length; i++) {
    obj[name[i]] = yearning[i];
  }

  for (let j = 0; j < photo.length; j++) {
    let count = 0;

    for (let k = 0; k < photo[j].length; k++) {
      count += obj[photo[j][k]] || 0;
    }

    result.push(count);
  }

  return result;
}

File: python/programmers/level.1/prgm_176963_1.py
def solution(name, yaerning, photo):
    score = {}
    answer = []

    for k, v in zip(name, yaerning):
        score[k] = v

    for i in photo:
        count = 0
        
        for j in i:
            if j in score: count += score[j]
        
        answer.append(count)

    return answer
    